bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/AbstractExpressionPart.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.common;
+
+import java.text.ParseException;
+import java.util.Collection;
+import java.util.TreeSet;
+
+/**
+ * <code>AbstractExpressionPart</code> is an abstract implementation of {@link ExpressionPart} that capture the common
+ * functionality to all ExpressionPart implementations.
+ *
+ * @author Karel Goderis - Initial contribution
+ *
+ */
+abstract class AbstractExpressionPart implements ExpressionPart {
+
+    private String part;
+    private BoundedIntegerSet valueSet;
+
+    public AbstractExpressionPart(String s) throws ParseException {
+        this.set(s);
+        this.parse();
+    }
+
+    @Override
+    public void set(String s) {
+        this.setPart(s);
+    }
+
+    /**
+     * @return the set of values that are valid for the expression part
+     */
+    public BoundedIntegerSet getValueSet() {
+        return valueSet;
+    }
+
+    /**
+     * @param set the set of values that the epxression part should take into consideration
+     */
+    public void setValueSet(BoundedIntegerSet set) {
+        this.valueSet = set;
+    }
+
+    abstract BoundedIntegerSet initializeValueSet();
+
+    /**
+     * A <code>BoundedIntegerSet</code> is a set of Integers that is bound by a minimum and maximum value, that does
+     * allow or not allow negaive values, and that is either starts from index 0 or 1
+     *
+     * @author Karel Goderis
+     *
+     */
+    protected class BoundedIntegerSet extends TreeSet<Integer> {
+        private static final long serialVersionUID = 19296179649170335L;
+        protected final int max;
+        protected final int min;
+        protected final boolean negative;
+        protected boolean is1indexed;
+
+        BoundedIntegerSet(final int min, final int max, final boolean negativeValuesAllowed, final boolean is1indexed) {
+            this.min = min;
+            this.max = max;
+            this.negative = negativeValuesAllowed;
+            this.is1indexed = is1indexed;
+        }
+
+        /**
+         * Validates if an integer can be added to the set
+         *
+         * @param integer - the integer to validate
+         */
+        private void validate(final Integer integer) {
+            if (negative) {
+                final int abs = Math.abs(integer);
+                if (abs < min || abs > max) {
```

Comment:
I believe I do not understand this class at a first look.
Reading this lines of code I assume that min and max must be >= 0.
If min=5 and max=7 and negative is true, then -6 is an allowed integer value?